src/Workspaces/Core/Portable/Workspace/Solution/FileTextLoader.cs

```diff
@@ -171,29 +172,21 @@ public override async Task<TextAndVersion> LoadTextAndVersionAsync(LoadTextOptio
             // this logic. This is tracked by https://github.com/dotnet/corefx/issues/6007, at least in
             // corefx. We also open the file for reading with FileShare mode read/write/delete so that
             // we do not lock this file.
-            using (var stream = FileUtilities.RethrowExceptionsAsIOException(() => new FileStream(Path, FileMode.Open, FileAccess.Read, FileShare.ReadWrite | FileShare.Delete, bufferSize: 1, useAsync: true)))
-            {
-                var version = VersionStamp.Create(prevLastWriteTime);
 
-                // we do this so that we asynchronously read from file. and this should allocate less for IDE case. 
-                // but probably not for command line case where it doesn't use more sophisticated services.
-                using var readStream = await SerializableBytes.CreateReadableStreamAsync(stream, cancellationToken: cancellationToken).ConfigureAwait(false);
-                var text = CreateText(readStream, options, cancellationToken);
-                textAndVersion = TextAndVersion.Create(text, version, Path);
-            }
+            var textAndVersion = await FileUtilities.RethrowExceptionsAsIOExceptionAsync(
```

Comment:
primary changes are:

1. RethrowExceptionsAsIOExceptionAsync now surrounds *all* the IO (including the deferred work done in CreateReadableStreamAsync).
2. this was converted to static-lambda form just to be a slightly better citizen about allocatinos.